lib/core/util.js

```diff
@@ -26,49 +27,12 @@ function isBlobLike (object) {
   )
 }
 
-function isObject (val) {
-  return val !== null && typeof val === 'object'
-}
-
-// this escapes all non-uri friendly characters
-function encode (val) {
-  return encodeURIComponent(val)
-}
-
-// based on https://github.com/axios/axios/blob/63e559fa609c40a0a460ae5d5a18c3470ffc6c9e/lib/helpers/buildURL.js (MIT license)
 function buildURL (url, queryParams) {
   if (url.includes('?') || url.includes('#')) {
     throw new Error('Query params cannot be passed when url already contains "?" or "#".')
   }
-  if (!isObject(queryParams)) {
-    throw new Error('Query params must be an object')
-  }
-
-  const parts = []
-  for (let [key, val] of Object.entries(queryParams)) {
-    if (val === null || typeof val === 'undefined') {
-      continue
-    }
-
-    if (!Array.isArray(val)) {
-      val = [val]
-    }
 
-    for (const v of val) {
-      if (isObject(v)) {
-        throw new Error('Passing object as a query param is not supported, please serialize to string up-front')
-      }
-      parts.push(encode(key) + '=' + encode(v))
-    }
-  }
-
-  const serializedParams = parts.join('&')
-
-  if (serializedParams) {
-    url += '?' + serializedParams
-  }
-
-  return url
+  return url + '?' + stringify(queryParams)
```

Comment:
What happens if stringify returns empty string? Can you handle that scenario?

Reply:
Do we need to check if the url already contains a query string?

Reply:
We don't need to do that since we already do that at the beginning of the string by checking for `?` or `#`.

Reply:
an empty string is handled now 👍